openlcs/packages/views.py

```diff
@@ -912,6 +912,11 @@ def post(self, request, *args, **kwargs):
 
         if serializer_data and len(serializer_data) == 1:
             obj = serializer_data[0]
+            source_ready = 'source' in obj and \
+                obj['source'] is None and parent != ''
```

Comment:
How about adding 'source' into component filter?

Reply:
Actually, the current filter field:name, version, release, arch, type is used to identify a component in our database. add `source` may be confusing? So I think it seems to be better that we judge `source` exists or not after we query the data record?

Reply:
We could talk more later about the component schema. According to a workaround, it is LGTM.

Reply:
np, since it is blocking the release, feel free to close it if the testing is passed.